eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -130,6 +130,10 @@ jobs:
       patterns: '*.tar.gz'
```

Comment:
It looks like we are downloading 4 tarballs only which two are necessary.  Have you considered filtering out or being more explicit on which artifacts are downloaded versus the new remove step?  This would be more efficient.

  patterns: |
    **/Private.SourceBuilt.Artifacts.*.tar.gz
    **/dotnet-sdk.-*.tar.gz